waku/v2/node/wakunode2.nim

```diff
@@ -393,9 +393,9 @@ proc mountFilter*(node: WakuNode) =
 
 # NOTE: If using the swap protocol, it must be mounted before store. This is
 # because store is using a reference to the swap protocol.
-proc mountSwap*(node: WakuNode) =
+proc mountSwap*(node: WakuNode, swapConfig: SwapConfig = SwapConfig(mode: SwapMode.Soft, paymentThreshold: 100, disconnectThreshold: -100)) =
```

Comment:
One thing you may still want to do is provide an `init*()`-function for `SwapConfig` where you set the default values  (somewhat like a default constructor). That way users of the type can choose to only set the config fields they're specifically interested in and leave the rest as default. See [this example](https://github.com/status-im/nim-libp2p/blob/master/libp2p/protocols/pubsub/gossipsub.nim#L41-L71) for `GossipSubParams` in libp2p.
This can be in a subsequent PR though.

Reply:
Thanks for pointing out. I have made the update and learnt something new as well.